The ``100-matrix_mul.py`` module
================================

Using the ``matrix_mul``
------------------------

First import ``matrix_mul``:

	>>> matrix_mul = __import__('100-matrix_mul').matrix_mul

Now use it:

	>>> print(matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]]))
	[[7, 10], [15, 22]]
	>>> print(matrix_mul([[1, 2]], [[3, 4], [5, 6]]))
	[[13, 16]]
	>>> print(matrix_mul([[1, 2], [3, 4], [5, 6]], [[1, 2, 3, 4], [5, 6, 7, 8]]))
	[[11, 14, 17, 20], [23, 30, 37, 44], [35, 46, 57, 68]]
	>>> print(matrix_mul("hello", [[1], [2]]))
	Traceback (most recent call last):
		...
	TypeError: m_a must be a list
	>>> print(matrix_mul([[1], [2]], "hello"))
	Traceback (most recent call last):
		...
	TypeError: m_b must be a list
	>>> print(matrix_mul([1, 2], [[1], [2]]))
	Traceback (most recent call last):
		...
	TypeError: m_a must be a list of lists
	>>> print(matrix_mul([[1], [2]], [1, 2]))
	Traceback (most recent call last):
		...
	TypeError: m_b  must be a list of lists
	>>> print(matrix_mul([], [[1], [2]]))
	Traceback (most recent call last):
		...
	ValueError: m_a can't be empty
	>>> print(matrix_mul([[]], [[1], [2]]))
	Traceback (most recent call last):
		...
	ValueError: m_a can't be empty
	>>> print(matrix_mul([[1], [2]], []))
	Traceback (most recent call last):
		...
	ValueError: m_b can't be empty
	>>> print(matrix_mul([[1], [2]], [[]]))
	Traceback (most recent call last):
		...
	ValueError: m_b can't be empty
	>>> print(matrix_mul([[1], ["#"]], [[1, 2]]))
	Traceback (most recent call last):
		...
	TypeError: m_a should contain only integers or floats
	>>> print(matrix_mul([[1], [2.5]], [[1, 2]]))
	[[1, 2], [2.5, 5.0]]
	>>> print(matrix_mul([[1], [2]], [[1, "#"]]))
	Traceback (most recent call last):
		...
	TypeError: m_b should contain only integers or floats
	>>> print(matrix_mul([[1], [2]], [[1, 2.5]]))
	[[1, 2.5], [2, 5.0]]
	>>> print(matrix_mul([[1], [2, 3]], [[1, 2.5]]))
	Traceback (most recent call last):
		...
	TypeError: each row of m_a must be of the same size
	>>> print(matrix_mul([[1], [2]], [[1, 2.5], [3]]))
	Traceback (most recent call last):
		...
	TypeError: each row of m_b must be of the same size
	>>> print(matrix_mul([[1, 2]], [[1, 2]]))
	Traceback (most recent call last):
		...
	ValueError: m_a and m_b can't be multiplied
